sway/config.c

```diff
@@ -560,16 +560,16 @@ static char *expand_line(const char *block, const char *line, bool add_brace) {
 
 bool read_config(FILE *file, struct sway_config *config) {
 	bool reading_main_config = false;
-	char *this_config = NULL, *config_pos;
-	long config_size = 0;
+	char *this_config = NULL;
+	unsigned long config_size = 0;
 	if (config->current_config == NULL) {
 		reading_main_config = true;
 
 		fseek(file, 0, SEEK_END);
 		config_size = ftell(file);
```

Comment:
While you're at it, can you add error handling for this?

Reply:
Added some, though not sure if it's good.